clang/test/SemaSYCL/intel-reqd-work-group-size-device.cpp

```diff
@@ -1,67 +0,0 @@
-// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2017 -Wno-sycl-2017-compat -fsyntax-only -verify %s
```

Comment:
Note to reviewers and myself: we have a positive test that checks that the attribute can be applied to lambdas and functors without any diagnostics in `clang/test/SemaSYCL/reqd-sub-group-size.cpp`. This test seems to be focused on propagation of the attribute from device functions to kernels which is not a thing in SYCL 2020 and therefore we don't need this test anymore.